napari/utils/io.py

```diff
@@ -1,16 +1,85 @@
 import os
+import re
 
 from glob import glob
 from pathlib import Path
 
 import numpy as np
-from skimage import io
-from skimage.io.collection import alphanumeric_key
 
 from dask import delayed
 from dask import array as da
 
 
+def imsave(filename: str, data: np.ndarray):
+    """custom imaplementation of imread to avoid skimage dependecy"""
+    ext = os.path.splitext(filename)[1]
+    if ext in [".tif", "tiff"]:
+        import tifffile
+
+        tifffile.imsave(filename, data)
+    else:
+        import imageio
+
+        imageio.imsave(filename, data)
+
+
+def imread(filename: str):
+    """custom imaplementation of imread to avoid skimage dependecy"""
+    ext = os.path.splitext(filename)[1]
+    if ext in [".tif", "tiff", ".lsm"]:
+        import tifffile
+
+        image = tifffile.imread(filename)
+    else:
+        import imageio
+
+        image = imageio.imread(filename)
+
+    if not hasattr(image, 'ndim'):
+        return image
+
+    if image.ndim > 2:
+        if image.shape[-1] not in (3, 4) and image.shape[-3] in (3, 4):
+            image = np.swapaxes(image, -1, -3)
+            image = np.swapaxes(image, -2, -3)
+    return image
+
+
+if not os.environ.get("NO_SKIMAGE", False):
+    try:
+        from skimage.io import imread as _imread
+    except ImportError:
+        _imread = imread
+else:
+    _imread = imread
```

Comment:
And then I would remove this logic as well, because otherwise, yikes. LOL Let's just only support tifffile and imageio, and if we miss out on something, then we can write an skimage io plugin.

Reply:
I do not understand. How writing skimage io plugin can help if  skimage support was removed. Yo mean skimage plugin support? skimage plugin porting?

I removed this.

Reply:
Sorry, I meant, we can write a napari IO plugin (see #1030) that uses skimage.io to read files.